File: frontend/src/types/activity-log.ts
/**
 * Activity log entity interfaces.
 */
import type { BaseEntity } from "./base";

export type ActivityLogEventType = "config_change" | "action";
export type ActivityLogSeverity = "info" | "warning" | "error";

export interface ActivityLogEntry extends BaseEntity {
  timestamp: string;
  user: string;
  eventType: ActivityLogEventType;
  entityType: string;
  entityId?: number | null;
  message: string;
  result?: string | null;
  severity: ActivityLogSeverity;
}

export interface ActivityLogPage {
  items: ActivityLogEntry[];
  total: number;
  page: number;
  pageSize: number;
  pages: number;
}

export interface ActivityLogFilters {
  dateFrom?: string;
  dateTo?: string;
  eventType?: ActivityLogEventType | "";
  entityType?: string;
  user?: string;
  severity?: ActivityLogSeverity | "";
  page?: number;
  pageSize?: number;
}

File: custom_components/device_manager/persistence/models/activity_log.py
"""DmActivityLog dataclass model.

Represents a single audit/activity log entry persisted in dm_activity_log.
"""

from dataclasses import dataclass, field
from typing import Optional

from .base import SerializableMixin


@dataclass
class DmActivityLog(SerializableMixin):
    """A single audit log entry recording a config change or triggered action.

    Fields:
        id:          Auto-increment primary key.
        timestamp:   UTC timestamp string (ISO 8601).
        user:        HA display name of the user who triggered the event.
        event_type:  'config_change' or 'action'.
        entity_type: Affected entity kind (device, room, floor, building,
                     device_model, device_firmware, device_function,
                     setting, import, deploy, scan, maintenance, ha_sync).
        entity_id:   FK to the affected record (nullable).
        message:     Human-readable description (supports Markdown).
        result:      Raw output / error log (nullable).
        severity:    'info', 'warning', or 'error'.
    """

    id: Optional[int] = field(default=None)
    timestamp: str = field(default="")
    user: str = field(default="system")
    event_type: str = field(default="action")
    entity_type: str = field(default="")
    entity_id: Optional[int] = field(default=None)
    message: str = field(default="")
    result: Optional[str] = field(default=None)
    severity: str = field(default="info")
